### PR TITLE
Ref/design system/footer

### PR DESCRIPTION
**Pull request by alexanderleegs**
## Problem

This PR refactors our existing Footer component to fully use chakra components instead of styled divs.

Closes https://github.com/isomerpages/isomercms-frontend/issues/861

## Solution

Due to the variations between different footer types (variable number of buttons, button text etc), we have decided to keep the `Footer` component as a simple container, with each component which requires the footer specifying the actual footer content. The existing areas in which the footer is used have been refactored to use this new format, and the other footer styled divs in edit homepage, contact-us, and navbar have also been refactored to use this component.

### CODE REVIEW

File: src/components/Footer/Footer.stories.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.stories.tsx:19]: nit: extra comment

File: src/components/Footer/Footer.stories.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.stories.tsx:34]: nit: extra comment

File: src/components/Footer/Footer.stories.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.stories.tsx:27]: could we change the name to avoid mixing numbers/letters in names

File: src/components/Footer/Footer.stories.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.stories.tsx:36]: we don't need to wrap in a fragment

File: src/components/Footer/Footer.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.tsx:9]: btw, we should specify this in rem probably. this is because rem is actually dynamically computed relative to the root element and allows responsive sizing (see [here](https://chiamakaikeanyi.dev/sizing-in-css-px-vs-em-vs-rem/))

File: src/components/Footer/Footer.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.tsx:14]: can keep border as px fyi

File: src/components/Footer/Footer.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.tsx:21]: nit: don't need to wrap in fragment

File: src/components/Footer/Footer.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.tsx:20]: just double checking here - is the intention for the children of `HStack` to be treated as a singular unit and justified to the end of the footer with 1rem spacing? 

this is the current behaviour and if design shows that the footer is consistent liek this across all pages then it's fine

File: src/components/Footer/Footer.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.tsx:4]: could we use `PropsWithChildren` from react here for btr typing?

**Review by seaerchin — COMMENTED** [review]
minor comments but overall looks fine!

**Comment by seaerchin** [comment]
code-wise seems ok, please wait for @NatMaeTan to approve design changes before merging!

File: src/components/Footer/Footer.stories.tsx
Comment (alexanderleegs) [reply src/components/Footer/Footer.stories.tsx]: Thanks for catching, https://github.com/isomerpages/isomercms-frontend/pull/932/commits/701f438ffdf74c5dc631deb36a01f621f694d103

File: src/components/Footer/Footer.stories.tsx
Comment (alexanderleegs) [reply src/components/Footer/Footer.stories.tsx]: https://github.com/isomerpages/isomercms-frontend/pull/932/commits/701f438ffdf74c5dc631deb36a01f621f694d103

File: src/components/Footer/Footer.stories.tsx
Comment (alexanderleegs) [reply src/components/Footer/Footer.stories.tsx]: Changed in https://github.com/isomerpages/isomercms-frontend/pull/932/commits/88f6867043db46c09fcb43abfad99a66a315a766!

File: src/components/Footer/Footer.stories.tsx
Comment (alexanderleegs) [reply src/components/Footer/Footer.stories.tsx]: Right, I was having issues with this earlier because of the props in Footer - but swapping to `PropsWithChildren` fixed the issue! https://github.com/isomerpages/isomercms-frontend/pull/932/commits/7faa682fd7af5140926947afef3520d12daa67fb

File: src/components/Footer/Footer.tsx
Comment (alexanderleegs) [reply src/components/Footer/Footer.tsx]: https://github.com/isomerpages/isomercms-frontend/pull/932/commits/7faa682fd7af5140926947afef3520d12daa67fb

File: src/components/Footer/Footer.tsx
Comment (alexanderleegs) [reply src/components/Footer/Footer.tsx]: Thanks for the suggestion, changed in https://github.com/isomerpages/isomercms-frontend/pull/932/commits/7faa682fd7af5140926947afef3520d12daa67fb

**Review by alexanderleegs — COMMENTED** [review]
Tagging @audreytcy as well!

**Comment by seaerchin** [comment]
@alexanderleegs you gotta assign them (@audreytcy and @NatMaeTan) to be reviewers on chromatic (ui review -> assign reviewers on the RHS); i've done it for you here but just fyi

File: src/components/Footer/Footer.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.tsx:11]: i don't see a problem with this but just wanted to point out that fixed removes it from the document flow (see [here](https://www.kevinpowell.co/article/positition-fixed-vs-sticky/)). i think it might be worth asking designers if the page is guaranteed to provide space for footer (when it's positioned) or otherwise, using sticky to ensure that this element is accounted for/using grid layout to reserve space

File: src/components/Footer/Footer.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.tsx:6]: just double checking here - would this not work with `PropsWithChildren<{}>`? it's a very minor change so dw about it but curious

**Review by seaerchin — APPROVED** [review]

File: src/components/Footer/Footer.tsx
Comment (alexanderleegs) [reply src/components/Footer/Footer.tsx:6]: I guess it does too! But I was getting a linting error, and the suggestion for empty objects was to do something like this instead

File: src/components/Footer/Footer.tsx
Comment (seaerchin) [inline src/components/Footer/Footer.tsx:6]: what we can do here is to have a rest parameter `...rest` and spread it after; this is so that the API stays identical to `Flex`

**Review by seaerchin — COMMENTED** [review]
minor nit on props

File: src/components/Footer/Footer.tsx
Comment (alexanderleegs) [reply src/components/Footer/Footer.tsx]: https://github.com/isomerpages/isomercms-frontend/pull/932/commits/c62f356a185d6ef76426aee118446b04d78f9e46

**Review by seaerchin — APPROVED** [review]

**Pull request merged**
